File: src/main/java/de/ingrid/igeserver/exports/iso/ResponsibleParty.java
package de.ingrid.igeserver.exports.iso;

import javax.xml.bind.annotation.XmlAttribute;
import javax.xml.bind.annotation.XmlElement;

@SuppressWarnings("unused")
public class ResponsibleParty {
	
	@XmlAttribute
	public String uuid;

	private String individualName;
	
	private String organisationName;
	
	private String positionName;
	
	@XmlElement(name = "contactInfo")
	public ContactInfo contactInfo;
	
	@XmlElement(name = "role")
	private RoleCode role;

	public void setRole(CodelistAttributes codelistAttr) {
		this.role = new RoleCode();
		this.role.codelist = codelistAttr;
	}

}

File: server/src/main/java/de/ingrid/igeserver/exports/iso/ResponsibleParty.kt
package de.ingrid.igeserver.exports.iso

import com.fasterxml.jackson.dataformat.xml.annotation.JacksonXmlProperty

data class ResponsibleParty(
    @JacksonXmlProperty(isAttribute = true) var uuid: String? = null,
    val individualName: CharacterString? = null,
    val organisationName: CharacterString? = null,
    val positionName: CharacterString? = null,
    var contactInfo: ContactInfo? = null,
    var role: RoleCode
)